### PR TITLE
BP: add avro.decode

### PR DESCRIPTION
**Pull request by hariso**
### Description

Please include a summary of the change and what type of change it is (new feature, bug fix, refactoring, documentation).
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes # (issue)

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/plugin/processor/builtin/avro/decode.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		return v, nil
+	case []byte:
+		return v, nil
+	case string:
+		decoded := make([]byte, base64.StdEncoding.DecodedLen(len(v)))
+		n, err := base64.StdEncoding.Decode(decoded, []byte(v))
+		if err != nil {
+			return nil, cerrors.Errorf("couldn't base64-decode data: %w", err)
+		}
+		return opencdc.RawData(decoded[:n]), nil
```
Comment (lovromazgon) [inline pkg/plugin/processor/builtin/avro/decode.go:198]: Hmm, so the assumption is that the string is coming from a JSON encoded record? 🤔 Is there a use case when this string wouldn't be base64 encoded?

File: pkg/plugin/processor/builtin/avro/decode.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+func (p *decodeProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "avro.encode",
+		Summary: "Decodes a field's raw data in the Avro format",
+		Description: `The processor takes raw data (bytes or a Base64-encoded string) in the specified field and decodes 
+it from the [Avro format](https://avro.apache.org/) into structured data. It extracts the schema ID from the data, 
+downloads the associated schema from the [schema registry](https://docs.confluent.io/platform/current/schema-registry/index.html)
+and decodes the payload. The schema is cached locally after it's first downloaded. 
+
+Currently, the processor only supports the Avro format. If the processor encounters structured data or the data
```
Comment (lovromazgon) [inline pkg/plugin/processor/builtin/avro/decode.go:118]: Well yeah, it's the `avro.decode` processor 😄 This is probably taken from the old processor, which was a "generic" processor.
```suggestion
If the processor encounters structured data or the data 
```

**Review by lovromazgon — COMMENTED** [review]
We need an example and this should be good to go!

File: pkg/plugin/processor/builtin/avro/decode.go (hunk 2 above)
Comment (hariso) [reply pkg/plugin/processor/builtin/avro/decode.go:118]: Ah, exactly!

File: pkg/plugin/processor/builtin/avro/decode.go (hunk 1 above)
Comment (lovromazgon) [reply pkg/plugin/processor/builtin/avro/decode.go]: We decided to put this functionality in a separate processor [base64 encode/decode](https://github.com/ConduitIO/conduit/pull/1412).

**Review by lovromazgon — APPROVED** [review]
LGTM!

File: pkg/plugin/processor/builtin/avro/decode.go (hunk 1 above)
Comment (hariso) [reply pkg/plugin/processor/builtin/avro/decode.go]: For the record, Lovro and I chatted about this. While it's likely that a string that represents Avro bytes is a Base64 encoding, it doesn't have to be. However, that's not always the case, meaning the processor will not always be usable out of the box. We will provide a Base64 encoder and decoder that can be used in these case.

**Pull request merged**